### ISSUE TITLE
Multiselet calendar don't update ngModel on unselect date

### ISSUE THREAD

**Issue by sgodoy** [primefaces/primeng#3876]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
```

**Plunkr Case (Bug Reports)**
Please fork the plunkr below and create a case demonstrating your bug report. Issues without a plunkr have much less possibility to be reviewed.

http://plnkr.co/edit/vG1QBJQuTVyEbhQRyM7s?p=preview

**Current behavior**
Multiselet calendar don't update ngModel when unselect date, until I select another one.

**Expected behavior**
Unselected date must update ngModel

**Minimal reproduction of the problem with instructions**

http://plnkr.co/edit/vG1QBJQuTVyEbhQRyM7s?p=preview

1- Select two or more dates 
2- Unselect a date
3- Displayed dates don't update

**What is the motivation / use case for changing the behavior?**
Can't unselect dates if I not select another one after

**Please tell us about your environment:**
Windows 7, Visual Studio Code.

* **Angular version:** 
^4.0.0

* **PrimeNG version:** 
^4.2.0-rc.1

* **Browser:** 
Chrome 60.0.3112.113 (64 bits)
Firefox 55.0.3 (32-bit)
 
* **Language:** 
TypeScript ~2.3.3

* **Node (for AoT issues):** `node --version` =    8.2.1

**Comment by cagataycivici** [comment]
I'm unable to replicate the issue. Feel free to attach a screencast.

**Comment by DRSchallerSS** [comment]
I've been waiting for this issue to be resolved as I can also replicate the issue in my project. The link to the plunker in the original issue message above clearly replicates the issue.
This issue should not be closed

Here are the steps to replicate the issue of the model not updating:
- Open the Plunker link
1. Click the calendar icon
2. Click multiple dates on the calendar.
2a. Note that the dates are displayed behind the calendar ( {{fechas}} ), clicking off to hide the calendar will show the text of the selected dates as they are slightly hidden by the calendar
3. Open the calendar and deselect the dates highlighted
3a. Note that the calendar dates displayed on the screen **do not change**.
4. Open the calendar, select a non selected date, the model updates and the dates displayed are changed.

**Comment by sgodoy** [comment]
Hi, here is a gif replicating the problem, it follow the steps given by @DRSchallerSS 

![GIF replicating the problem](https://user-images.githubusercontent.com/1865840/31893173-3c458e86-b7e1-11e7-8dd6-ae234aad44c6.gif)

**Comment by fusionart** [comment]
I also have the same issue as **sgodoy** is showning on the gif.

**Comment by DRSchallerSS** [comment]
Does a new issue have to be opened to have this addressed?

**Comment by med-doulfakar** [comment]
Am always having the same issue ! is there any fix ?

**Comment by gklprsna** [comment]
I am also facing same issue, please anyone fix this!!

**Comment by xyrintech** [comment]
Issue is still there.

May I know which version this is being resolved in?
I am using 4.1.3

**Comment by xyrintech** [comment]
I have tried 4.3.0 and this issue is till there

**Comment by thanhnguyen92** [comment]
This issue won't fix on 4.3.0. It just fixed from 5.2.0-rc2
